Source/WebCore/rendering/RenderBlock.cpp

```diff
@@ -3548,4 +3548,14 @@ LayoutUnit RenderBlock::layoutOverflowLogicalBottom(const RenderBlock& renderer)
     return std::max(renderer.clientLogicalBottom(), maxChildLogicalBottom + renderer.paddingAfter());
 }
 
+void RenderBlock::updateDescendantTransformsAfterLayout()
+{
+    auto descendants = view().frameView().layoutContext().descendantsNeedingTransformUpdate(this);
```

Comment:
Rather than fetching `descendantsNeedingTransformUpdate` and then clearing later via `markFinishedLayout`, you could just have a single `take` function that uses `std::exchange()` to clear the hash map and return it.

Reply:
Needed due to above.

Reply:
It could still be `takeBoxesNeedingTransformUpdateAfterContainerLayout()` which would extract them from the hash map, and remove the need for `finishedUpdatingDescendants()`.

Reply:
This was originally to prevent us from erroneously adding the `RenderBox` back into the vector when we call `updateTransform` from in `updateDescendantTransformsAfterLayout`. However, we can get around this by directly calling `updateTransform` on the layer, bypassing the code in `RenderLayerModelObject::updateLayerTransform` that would incorrectly re-add it. This helps simplify the code by using take and removing `finishedUpdatingDescendants`.